### PR TITLE
Add Playwright Compatibility-classic-editor Test

### PR DESCRIPTION
**Pull request by alvitazwar**
## What?
Based on https://github.com/WordPress/gutenberg/pull/38570, part of https://github.com/WordPress/gutenberg/issues/38851. Migrate [compatibility-classic-editor.test.js](https://github.com/alvitazwar/gutenberg/blob/add/playwright-compatibility/test/e2e/specs/editor/various/compatibility-classic-editor.spec.js) to its Playwright counterpart.

## Why?
See https://github.com/WordPress/gutenberg/pull/38570 for its background and rationale.

## How?
See https://github.com/WordPress/gutenberg/pull/38570 for the proposed migration steps.

## Testing Instructions
npm run test:e2e:playwright /test/e2e/specs/editor/various/compatibility-classic-editor.spec.js

## Screencast

https://user-images.githubusercontent.com/55917380/193242117-aba04bc4-2bca-4105-94ef-af69e08f5601.mov

### CODE REVIEW

**Comment by alvitazwar** [comment]
@ntwb, @nerrad  can you please review this?

**Comment by alvitazwar** [comment]
@kevin940726 Can you please review this PR?

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	test( 'Should not apply autop when rendering blocks', async ( {
+		page,
+		editor,
+	} ) => {
+		// Insert Html
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/html' );
+		await expect(
+			page.locator( 'role=option[name="Custom HTML"i][selected]' )
+		).toBeVisible();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: Let's use `editor.insertBlock` if possible?

```suggestion
		await editor.insertBlock( { name: 'core/html' } );
		await page.focus( 'role=textbox[name="HTML"i]' );
```

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+		await page.keyboard.press( 'Enter' );
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: Why do we need this Enter?

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		await page.keyboard.type( '<a>' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Random Link' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '</a> ' );
+		// Publish Post
+		await editor.publishPost();
+
+		// View Post
+		await page.locator( 'a.components-button.is-primary' ).click();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: Let's use role selectors to increase readability and confidence of tests.

```suggestion
		await page.click( 'role=link[name="View post"i]' );
```

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+		// Check the content doesn't contain <p> tags.
+		await page.waitForSelector( '.entry-content' );
+		const content = await page.$eval( '.entry-content', ( element ) =>
+			element.innerHTML.trim()
+		);
+		expect( content ).toMatchSnapshot();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: It's a bit unfortunate that we don't have accessible selectors for `.entry-content` for now 🤔 . However, I think we can just assert the output as Playwright will implicitly wait for the element. Also, I think we don't need the snapshot.

```suggestion
		// No accessible selector for now.
		const content = page.locator( '.entry-content' );
		await expect.poll( () => content.innerHTML() ).toBe(`<a>
Random Link
</a>`);
```

**Review by kevin940726 — COMMENTED** [review]
Thanks for working on this! Sorry for the late review 🙇

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js (hunk 2 above)
Comment (alvitazwar) [reply test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: No, I removed all the entries which seem to be unnecessary.

**Comment by alvitazwar** [comment]
@kevin940726  can you please check now?

**Comment by alvitazwar** [comment]
@kevin940726  please review once, whenever you get time :)

File: test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Create-Columns-and-verify-1-chromium.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<a>
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Create-Columns-and-verify-1-chromium.txt]: I think this file can now be deleted?

File: test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Should-not-apply-autop-when-rendering-blocks-1-chromium.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<a>
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Should-not-apply-autop-when-rendering-blocks-1-chromium.txt]: This as well.

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	} ) => {
+		await editor.insertBlock( { name: 'core/html' } );
+		await page.focus( 'role=textbox[name="HTML"i]' );
+		await page.keyboard.type( '<a>' );
+		await page.keyboard.type( 'Random Link' );
+		await page.keyboard.type( '</a> ' );
+		// Publish Post
+		await editor.publishPost();
+		// View Post
+		await page.locator( 'role=link[name="View post"i]' ).first().click();
```
Comment (kevin940726) [inline test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: Avoid using `first()`, trying chaining selectors instead.

```suggestion
		await page.click( 'role=region[name="Editor publish"i] >> role=link[name="View post"i]' );
```

File: test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Create-Columns-and-verify-1-chromium.txt (hunk 5 above)
Comment (alvitazwar) [reply test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Create-Columns-and-verify-1-chromium.txt]: Done

File: test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Should-not-apply-autop-when-rendering-blocks-1-chromium.txt (hunk 6 above)
Comment (alvitazwar) [reply test/e2e/specs/editor/various/__snapshots__/Compatibility-with-classic-editor-Should-not-apply-autop-when-rendering-blocks-1-chromium.txt]: Done

File: test/e2e/specs/editor/various/compatibility-classic-editor.spec.js (hunk 7 above)
Comment (alvitazwar) [reply test/e2e/specs/editor/various/compatibility-classic-editor.spec.js]: Addressed. Thanks

**Review by kevin940726 — APPROVED** [review]
LGTM 👍 Thanks!

**Pull request merged**